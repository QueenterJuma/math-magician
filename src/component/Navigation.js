import React from 'react';
import { Link } from 'react-router-dom';

const Navigation = () => (
  <div>
    <nav className="nav">
      <a href="/" className="title">
        Math Magician
      </a>
      <ul>
        <li>
          <Link to="/" className="nav-link">
            Home
          </Link>
        </li>
        <li>
          <Link to="/Calculator" className="nav-link">
            Calculator
          </Link>
        </li>
        <li>
          <Link to="/Quote" className="nav-link">
            Quote
          </Link>
        </li>
      </ul>
    </nav>
  </div>
);

export default Navigation;
