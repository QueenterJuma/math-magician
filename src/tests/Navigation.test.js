import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../Component/Navigation.js';

describe('Navigation', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
