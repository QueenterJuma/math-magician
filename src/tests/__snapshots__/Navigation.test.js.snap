// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div>
  <nav
    className="nav"
  >
    <a
      className="title"
      href="/"
    >
      Math Magician
    </a>
    <ul>
      <li>
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/Calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/Quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
